jax/api.py

```diff
@@ -2081,20 +2085,27 @@ def device_put_sharded(x: Sequence[Any], devices: Sequence[xc.Device]):
   - device_put
   - device_put_replicated
   """
-  if not isinstance(x, Sequence):
-    raise ValueError(f"x must be a sequence; got {type(x)}")
-  # TODO(jakevdp): provide a default for devices that considers both local devices and pods
-  assert len(x) == len(devices), f"len(x) = {len(x)} must equal len(devices) = {len(devices)}."
+  # TODO(jakevdp): provide a default for devices that considers both local
+  # devices and pods
+  if not isinstance(shards, Sequence):
+    raise ValueError("device_put_sharded `shards` input must be a sequence; "
+                     f"got {type(shards)}")
+  if not len(shards) == len(devices):
+    raise ValueError(f"len(shards) = {len(shards)} must equal "
+                     f"len(devices) = {len(devices)}.")
```

Comment:
I'm looking forward to the day when we're on Python 3.8 and can write this as `f"{len(shards)=} must equal {len(devices)=}"` :grin:

Reply:
![image](https://user-images.githubusercontent.com/1458824/101202433-1722af80-361e-11eb-9829-4ee5db6425c6.png)